src/pages/index.js

```diff
@@ -53,38 +71,87 @@ const profilePopup = new PopupWithForm(
   config.popupProfileSelector,
   (userData) => {
     userInfo.setUserInfo(userData);
-    profilePopup.close();
+    profilePopup.notifyUserSaving(true);
+    api.updateUserInfo(userData).finally(() => {
+      profilePopup.notifyUserSaving(false);
+      profilePopup.close();
+    });
   }
 );
 profilePopup.setEventListeners();
 
 const cardPopup = new PopupWithForm(config.popupAddCardSelector, (cardData) => {
-  const cardElement = createCard(cardData);
-  cardList.addItem(cardElement, false);
-  cardPopup.close();
+  cardPopup.notifyUserSaving(true);
+  api
+    .addCard(cardData)
+    .then((res) => {
+      cardData._id = res._id;
+      cardData.isLiked = false;
+      const cardElement = createCard(cardData);
+      cardList.addItem(cardElement, false);
+    })
+    .finally(() => {
+      cardPopup.notifyUserSaving(false);
+      cardPopup.close();
+    });
 });
 cardPopup.setEventListeners();
 
-const cardList = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      cardList.addItem(createCard(cardData));
-    },
-  },
-  config.cardsContainer
+const cardDeletePopup = new PopupWithForm(
+  config.popupDeleteCardSelector,
+  (cardData) => {
+    api.deleteCard(cardData).finally(() => {
+      cardDeletePopup.close();
+      cardToDelete.deleteCard();
+      cardToDelete = null;
+    });
+  }
 );
-cardList.renderItems();
+cardDeletePopup.setEventListeners();
+
+const changeAvatarPopup = new PopupWithForm(
+  config.popupChangeAvatarSelector,
+  (userData) => {
+    changeAvatarPopup.notifyUserSaving(true);
+    api.changeAvatar(userData).finally(() => {
+      changeAvatarPopup.notifyUserSaving(true);
```

Comment:
You need to start loading **berfore** the request 


But here you need `false` to stop loading

Reply:
done